### PR TITLE
upgrade cloudevents client.

### PR DESCRIPTION
**Pull request by morvencao**

### CODE REVIEW

File: internal/cloudevents/codec.go [hunk 1]
```diff
@@ -10,40 +10,37 @@ import (
...
+func (codec *Codec) Encode(source string, eventType cetypes.CloudEventsType, obj *db.Resource) (*cloudevents.Event, error) {
 	evt := cloudevents.NewEvent()
 	evt.SetID(uuid.New().String())
 	evt.SetSource(source)
 	evt.SetType(eventType.String())
 	evt.SetTime(time.Now())
 	evt.SetDataContentType("application/json")
-	evt.SetExtension("resourceID", obj.Id)
-	evt.SetExtension("resourceVersion", obj.ResourceGenerationID)
+	evt.SetExtension("resourceid", obj.Id)
```
Comment (qiujian16) [inline internal/cloudevents/codec.go]: nit I think "resourceid" should be a const in api.

File: internal/cloudevents/codec.go [hunk 2]
```diff
@@ -10,40 +10,37 @@ import (
...
 
 	if !obj.Object.GetDeletionTimestamp().IsZero() {
 		evt.SetExtension("deletionTimestamp", obj.Object.GetDeletionTimestamp().Time)
 		return &evt, nil
 	}
 
-	resourcePayload := &Resource{
-		Manifest: &obj.Object,
+	resourcePayload := &workpayload.Manifest{
+		Manifest: obj.Object,
```
Comment (qiujian16) [inline internal/cloudevents/codec.go:39]: Do we always need to set statusfeedback now?

File: internal/cloudevents/codec.go (hunk 1 above)
Comment (morvencao) [reply internal/cloudevents/codec.go]: updated this to use eventBuilder from API.

File: internal/cloudevents/codec.go (hunk 2 above)
Comment (morvencao) [reply internal/cloudevents/codec.go:39]: The work-client code for testing sets `statusfeedback`.
You're right, we should not relay on work-client to set `statusfeedback`,
I updated encode to set `statusfeedback` explicitly.

**Pull request merged**